core/panic-handler/src/lib.rs

```diff
@@ -39,30 +52,55 @@ This is a bug. Please report it at:
 ")}
 
 /// Set aborting flag. Returns previous value of the flag.
-pub fn set_abort(enabled: bool) -> bool {
+fn set_abort(enabled: bool) -> bool {
 	ABORT.with(|flag| {
 		let prev = flag.get();
 		flag.set(enabled);
 		prev
 	})
 }
 
-/// Abort flag guard. Sets abort on construction and reverts to previous setting when dropped.
-pub struct AbortGuard(bool);
+/// RAII guard for whether panics in the current thread should unwind or abort.
+///
+/// Sets a thread-local abort flag on construction and reverts to the previous setting when dropped.
+/// Does not implement `Send` on purpose.
+///
+/// > **Note**: Because we restore the previous value when dropped, you are encouraged to leave
+/// > the `AbortGuard` on the stack and let it destroy itself naturally.
+pub struct AbortGuard {
+	/// Value that was in `ABORT` before we created this guard.
+	previous_val: bool,
+	/// Marker so that `AbortGuard` doesn't implement `Send`.
+	_not_send: PhantomData<std::rc::Rc<()>>
```

Comment:
If you were looking for a simpler alternative, `*const ()` should [do](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=ae4e590a3c4674504a1eef97561747db).

Reply:
I was specifically looking for something that clearly doesn't implement `Send`

Reply:
> There's unfortunately no way AFAIK to write a test to make sure that it's not Send

Not to say that we should use it here, but:

https://docs.rs/static_assertions/0.3.4/static_assertions/macro.assert_not_impl_all.html